melusine/nlp_tools/lemmatizer.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import os
+from typing import Union
+
+import numpy
+import spacy
+from spacy.language import Language
+from spacy_lefff import LefffLemmatizer, POSTagger, Downloader
+
+from melusine.utils.verbitim_transformer import VerbitimTransformer
+
+
+@Language.factory('melt_tagger')
+def create_melt_tagger(nlp, name):
+    # TODO:Variable de release de l'asset
+    return POSTagger()
+
+
+@Language.factory('lefff_lemmatizer')
+def create_french_lemmatizer(nlp, name):
+    return LefffLemmatizer(after_melt=True, default=True)
+
+
+class BadConfigurationError(Exception):
+    """Exception to raise when a bad configuration was passed to lemmatization engine"""
+    pass
+
+
+class Lemmatizer(VerbitimTransformer):
+    """ Lemmatizer component for Socle NLP
+        Available engines include : spacy, Lefff and TreeTagger
+    """
+
+    _AVAILABLE_ENGINES = ['spacy',
+                          'Lefff'
+                          ]
+
+    _SPACY_ARGS = {'exclude': ['senter', 'ner']}
+
+    def __init__(self,
+                 in_col: str,
+                 out_col: str,
+                 engine: str,
+                 engine_conf: Union[dict, str, None],
+                 n_jobs: int = 1,
+                 batch_size: int = 1000
+                 ):
+        """
+        Initialize a Socle NLP Lemmatizer object
+        :param in_col:  input column of dataframe.
```

Comment:
Could you use numpy format for docstring? here is the link: https://numpydoc.readthedocs.io/en/latest/format.html